src/Control/Effect/Diagnostics.hs

```diff
@@ -98,6 +99,12 @@ fatalOnIOException ctx go = context ctx $ catch go die'
   where
     die' (e :: IOException) = fatalText ("io exception: " <> toText (show e))
 
+-- | Throw a generic error message on any exception, wrapped in a new 'context' using the provided @Text@.
+fatalOnSomeException :: (Has (Lift IO) sig m, Has Diagnostics sig m) => Text -> m a -> m a
+fatalOnSomeException ctx go = context ctx $ catch go die'
```

Comment:
use `Control.Exception.Extra.safeCatch` here

the short explanation is that we don't want to catch "async exceptions", like `Ctrl-C` -- because we'll end up continuing/ignoring it, which stops us from being able to terminate early with `Ctrl-C`

`safeCatch` is a wrapper around `catch` that ignores async exceptions